cilium/CFP-30984-dns-proxy-ha.md

```diff
@@ -94,6 +97,64 @@ Since policy revision numbers from Cilium are reset when the local Cilium instan
 
 ## Impacts / Key Questions
 
+### Getting the DNS Rules
+
+We need the DNS Rules for the Standalone DNS proxy to enforce the L7 DNS policy. The policy are the source of turth for the rules and are propagated to the agent when we apply the policy. We explore the options to get the DNS rules for the DNS proxy.
+
+#### Option 1: Running the GRPC sever in the agent[Recommended]
+
+We can run a gRPC server in the agent to serve the DNS rules to the DNS proxy. SDP will be responsible for creating the connection with the agent. And once SDP establish a connection agent can keep track of the stream and send the DNS rules to the SDP. Agent can then reuse the same stream to send updates in the DNS rules.
+In case, cilium agent is still not up, SDP will keep trying to connect to the agent until the connection is established. 
+
+##### Pros
+
+* SDP instances has the responsibility to connect to the agent.
+* Reusing the same stream will be effiecient in terms of resources as we are not creating/destroying the stream for every update.
+
+##### Cons
+
+* An overhead on the cilium agent to keep track of the streams of the connected SDP instances.
+* Streams are not thread safe, so if we have multiple threads using the same stream we will need to handle the synchronization.
+* If SDP in future decides to handle policy updates as well, it will be tricky. SDP needs to keep trying to sync with CA as well as CA needs to keep a track of the SDP instances to send updates.
+
+![standalone-dns-proxy-up-scenario](./images/standalone-dns-proxy-up-scenario.png)
+![standalone-dns-proxy-down-scenario](./images/standalone-dns-proxy-down-scenario.png)
+
+
+#### Option 2: Running the GRPC server in the SDP as well
```

Comment:
If option 1 is possible, do we really need option 2 ?

Reply:
We tested this option too as well.
I added it keeping in mind that SDP may/may not handle policy in the future. So we might need to consider a grpc server in the SDP.

Reply:
SDP not handling policy would be violate the toFQDN spec right ?

Reply:
Sorry, I am not sure if I understand this. What do you mean by violating the `toFQDN` spec ?